src/components/Main/Main.tsx

```diff
@@ -0,0 +1,42 @@
+import { useContext, useMemo, useState } from 'react';
+import { TodosContext } from '../../context/TodosContext';
+import { Todo } from '../../types/Todo';
+import { TodoList } from '../TodoList/TodoList';
+import { Status } from '../../enums/Status';
+
+export const Main = () => {
+  const { todos, updateTodos } = useContext(TodosContext);
+  const [isToggleAllChecked, setIsToggleAllChecked] = useState(true);
+
+  const isAllChecked = useMemo(() => {
+    return todos.every((todo: Todo) => todo.completed === Status.Completed);
+  }, [todos]);
+
+  const handleToggleAll = () => {
+    setIsToggleAllChecked(!isToggleAllChecked);
+
+    updateTodos(todos.map((todo: Todo) => ({
+      ...todo,
+      completed: isToggleAllChecked ? Status.Completed : '',
```

Comment:
The assignment of the 'completed' property is not correct. It should be a boolean value, not a string from the enum Status. If 'isToggleAllChecked' is true, then all todos should be marked as not completed (false), and vice versa.